homeassistant/components/ihc/auto_setup.py

```diff
@@ -86,7 +86,8 @@
 
 def autosetup_ihc_products(hass: HomeAssistant, config, ihc_controller, controller_id):
     """Auto setup of IHC products from the IHC project file."""
-    if not (project_xml := ihc_controller.get_project()):
+    project_xml = ihc_controller.get_project()
+    if not project_xml:
```

Comment:
This was updated in https://github.com/home-assistant/core/pull/58827 as part of https://github.com/home-assistant/architecture/discussions/617 so i don't think this should be reverted.

Reply:
Ok I was not aware of that - I have reverted it. 
(I do however find it a strange decision - in other languages I have worked with it considered bad practice to do assignment and condition in one because many don't see the assignment, only the condition)